tests/v2/validation/provisioning/rke2/hostname_truncation_test.go

```diff
@@ -46,123 +47,40 @@ func (r *HostnameTruncationTestSuite) SetupSuite() {
 }
 
 func (r *HostnameTruncationTestSuite) TestProvisioningRKE2Cluster() {
+	poolNames := permutations.GetTestName(r.clustersConfig.NodesAndRoles)
 	tests := []struct {
-		name                        string
-		machinePools                []provv1.RKEMachinePool
-		defaultHostnameLengthLimits []int
+		name                       string
+		poolLevelHostnameLength    []int
+		clusterLevelHostnameLength []int
 	}{
 		{
-			name: "Cluster level truncation",
-			machinePools: []provv1.RKEMachinePool{
-				{
-					Name: namegen.RandStringLower(1),
-				},
-				{
-					Name: namegen.RandStringLower(31),
-				},
-				{
-					Name: namegen.RandStringLower(63),
-				},
-			},
-			defaultHostnameLengthLimits: []int{0, 10, 31, 63},
-		},
-		{
-			name: "Machine pool level truncation - 10 characters",
-			machinePools: []provv1.RKEMachinePool{
-				{
-					Name:                namegen.RandStringLower(1),
-					HostnameLengthLimit: 10,
-				},
-				{
-					Name:                namegen.RandStringLower(31),
-					HostnameLengthLimit: 10,
-				},
-				{
-					Name:                namegen.RandStringLower(63),
-					HostnameLengthLimit: 10,
-				},
-			},
-			defaultHostnameLengthLimits: []int{10, 16, 63},
-		},
-		{
-			name: "Machine pool level truncation - 31 characters",
-			machinePools: []provv1.RKEMachinePool{
-				{
-					Name:                namegen.RandStringLower(1),
-					HostnameLengthLimit: 31,
-				},
-				{
-					Name:                namegen.RandStringLower(31),
-					HostnameLengthLimit: 31,
-				},
-				{
-					Name:                namegen.RandStringLower(63),
-					HostnameLengthLimit: 31,
-				},
-			},
-			defaultHostnameLengthLimits: []int{10, 31, 63},
-		},
-		{
-			name: "Machine pool level truncation - 63 characters",
-			machinePools: []provv1.RKEMachinePool{
-				{
-					Name:                namegen.RandStringLower(1),
-					HostnameLengthLimit: 63,
-				},
-				{
-					Name:                namegen.RandStringLower(31),
-					HostnameLengthLimit: 63,
-				},
-				{
-					Name:                namegen.RandStringLower(63),
-					HostnameLengthLimit: 63,
-				},
-			},
-			defaultHostnameLengthLimits: []int{10, 31, 63},
-		},
-		{
-			name: "Cluster and machine pool level truncation - 31 characters",
-			machinePools: []provv1.RKEMachinePool{
-				{
-					Name:                namegen.RandStringLower(1),
-					HostnameLengthLimit: 31,
-				},
-				{
-					Name: namegen.RandStringLower(31),
-				},
-				{
-					Name:                namegen.RandStringLower(63),
-					HostnameLengthLimit: 31,
-				},
-			},
-			defaultHostnameLengthLimits: []int{10, 31, 63},
+			name:                       "Cluster and Pool level truncation" + poolNames,
+			poolLevelHostnameLength:    []int{10, 31, 63},
+			clusterLevelHostnameLength: []int{10, 31, 63},
```

Comment:
I still doin't like how you changed this, it makes the test name super hard to follow and would like to be reverted to how it was before. I agree with what Jake said in registries.

Reply:
I still think the same thing here.